### ISSUE TITLE
can not record the self-defined topic using ros2 bag record

### ISSUE THREAD

**Issue by plane-li** [autowarefoundation/autoware.universe#812]
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I'm convinced that this is not my fault but a bug.

### Description

I can record the image and pointcloud throught" ros2 bag record --topic",however can not record the self-defined topic like"/perception/object_recognition/detection/apollo/objects".
Before command" ros2 bag record",I have done"source install/setup.bash " in the worksapce. At tha same time,the autoware is running,the network is ok and  "ros2 topic hz" is also usefull.

### Expected behavior

ros2 bag can record any topic in the related workspace!

### Actual behavior

It is always in the state of " Listening for topics..."
![1730722490](https://user-images.githubusercontent.com/103238330/165230126-c1582046-5b98-4127-ab80-b7f811bd40f4.jpg)

### Steps to reproduce

1.roslaunch autoware
2.open another terminal 
3.source install/setup.bash
4.ros2 bag record /perception/object_recognition/detection/apollo/objects

**Comment by BonoloAWF** [comment]
@kenji-miyake @angry-crab could you please check whether you are able to reproduce this bug? rosbag2 should be recording the available topics.

**Comment by kenji-miyake** [comment]
@plane-li I have two requests. :pray: 
- Could you provide the rosbag if you use some log data?
- Could you write the exact commands you executed?

@BonoloAWF I can try to reproduce this if he can provide more information.
I have one question and one request for you.
- Is this related to some projects such as Bus ODD?
- I think it's good to assign only one person to avoid double work and to make the driver clear.

**Comment by BonoloAWF** [comment]
@kenji-miyake  Noted, I've updated the assignees. It is indirectly related to the Bus ODD since this is a functionality we will need.

**Comment by angry-crab** [comment]
@kenji-miyake Sure! I'll look into it.

**Comment by plane-li** [comment]
Thank you very much!, We have solved it.
I have tried to use another computer to record the topic my computer pub, it is ok! 
So It may have some problems with the ros2 environment  of my computer.
Two steps to solve:
1.reinstall the rosbag.
   **command: sudo apt-get install ros-galactic-rosbag2***
2.rebuild the aw
  **command: colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release**
 
After the above two steps in my computer,  the ros2 bag can record available topics.
